src/react/components/common/cloudFilePicker/cloudFilePicker.tsx

```diff
@@ -0,0 +1,172 @@
+import React from "react";
+import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from "reactstrap";
+import { ICloudConnection, IConnection } from "../../../../models/applicationState";
+import { IStorageProvider, StorageProviderFactory } from "../../../../providers/storage/storageProvider";
+import CondensedList, { ListItem } from "../condensedList/condensedList";
+
+export interface ICloudFilePickerProps {
+    isOpen: boolean;
+    connections: IConnection[];
+    onCancel: () => void;
+    onSubmit: (content: string) => void;
+    fileExtension?: string;
+}
+
+export interface ICloudFilePickerState {
+    modalHeader: string;
+    condensedList: any;
+    selectedConnection: ICloudConnection;
+    selectedFile: string;
+    okDisabled: boolean;
+    backDisabled: boolean;
+}
+
+export function isCloudConnection(connection: IConnection): boolean {
```

Comment:
Instead of checking for the existence of specific properties I would add a new field/enum to the IStorageProvider interface where we can introduce a new property mapped to an enum like "StorageType": Local|Cloud|Other.

Then we can check this value.